blackjax/inference/trajectory.py

```diff
@@ -289,18 +280,29 @@ def add_one_state(expansion_state):
             _,
             step,
             proposal,
-            _,
             trajectory,
             termination_state,
             is_diverging,
             has_terminated,
         ) = jax.lax.while_loop(
             do_keep_integrating, add_one_state, initial_integration_state
         )
+        # In the while_loop we always extend on the right most direction.
+        new_trajectory = jax.lax.cond(
```

Comment:
@rlouf we can also fold this into `reorder_trajectories`, WDYT?

Reply:
I think that would make it uselessly complicated, we do not gain anything in terms of computation and it is a lot less explicit.